app/assets/javascripts/stores/message_store.js.coffee

```diff
@@ -0,0 +1,55 @@
+window.messageStore = Reflux.createStore
+  init: ->
+    @listenToMany(userActions)
+    @listenTo remoteActions.message, 'onReceiveMessage'
+    @state = @getInitialState()
+    @fetchMessages()
+    @listenForMessages()
+
+  getInitialState: ->
+    messages: []
+    loaded: false
+
+  fetchMessages: ->
+    $.ajax
+      url: "/messages"
+      dataType: 'json'
+      method: 'GET'
+    .done (data) =>
+      @state.messages = data
+      @state.loaded = true
+      @trigger @state
```

Comment:
Calling `trigger` on a store just causes it to trigger an update to anything listening to it.  In this case, the `Chat` component is listening through `messageStore.connect`, so any time we `trigger` here, `this.state.messages` in `Chat` gets updated to whatever we passed to `trigger`.